asmcomp/cmmgen.ml

```diff
@@ -1364,9 +1364,17 @@ let rec transl = function
         Cop(Capply(typ_addr, dbg), [get_field clos 0; transl arg; clos]))
   | Ugeneric_apply(clos, args, dbg) ->
       let arity = List.length args in
-      let cargs = Cconst_symbol(apply_function arity) ::
-        List.map transl (args @ [clos]) in
-      Cop(Capply(typ_addr, dbg), cargs)
+      bind "fun" (transl clos) (fun transl_clos ->
```

Comment:
Using `clos` rather than `trans_clos` as in the branch above (or using `trans_clos` above) would make it easier to compare the two `Ugeneric_apply` cases.